### PR TITLE
Feature/metamodule

### PR DESCRIPTION
**Pull request by speshoria**
- Added a new project for the metamodule.
- Updated the Packaging targets file and csproj file for not including settings, help file etc. in the package. 
- Added a new PowerShell script to cleanup the NuGet package references in the Nuspec (dropping the .NET Standard 2.0 Group) and update the modules with RequiredModules section

### CODE REVIEW

File: scripts/UpdateMetaModule.ps1 [hunk 1]
```diff
@@ -0,0 +1,82 @@
+[CmdletBinding()]
```
Comment (CodeCyclone) [inline scripts/UpdateMetaModule.ps1]: This script will be used more than just the metamodule. Can you rename it to UpdateModuleDependencies.ps1

File: scripts/UpdateMetaModule.ps1 [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+param
+(
+    [Parameter(Mandatory)]
+    [ValidateNotNullOrEmpty()]
+    [string] $Path
+)
+
+Write-Output "Running $($MyInvocation.MyCommand.Name)"
+
+$nuspecPath = (Get-Childitem �Path $Path -Include *.nuspec -Recurse).FullName
```
Comment (CodeCyclone) [inline scripts/UpdateMetaModule.ps1]: We should support multiple nuspecs and then iterate over them

Comment (CodeCyclone) [reply scripts/UpdateMetaModule.ps1]: Change Include to Filter

File: scripts/UpdateMetaModule.ps1 [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+Write-Output "Get all dependencies under .NET Standard 2.0 Group"
+$dep = $nuspecContent.package.metadata.dependencies.group.dependency
+if(!$dep)
+{
+    throw "No dependencies found in the nuspec file"
+}
+Write-Output "Found dependencies $dep"
+
+Write-Output "Removing the attributes which are not required and appending the dependencies"
+$dep | % { $_.RemoveAttribute("exclude"); $nuspecContent.package.metadata.dependencies.AppendChild($_) }
```
Comment (CodeCyclone) [inline scripts/UpdateMetaModule.ps1]: Don't you need the xmlns path too?

File: scripts/UpdateMetaModule.ps1 [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+$psdFileContent = Get-Content -Path $psdFilePath -Raw -ErrorAction Stop
+if(!$psdFileContent)
+{
+    throw "Module file is empty under path: $psdFilePath"
+}
+Write-Output "Found content for module file"
+$updatePsdFile = $false
+
+Write-Output "Getting list of all dependencies to add to module file Required Modules section"
+foreach ($dependency in $dep)
```
Comment (CodeCyclone) [inline scripts/UpdateMetaModule.ps1]: Do $dependencies = $dep -join ','

File: scripts/UpdateMetaModule.ps1 [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+{
+    $dependencies += " $($dependency.id),"
+}
+$dependencies = $dependencies.TrimEnd(",")
+
+
+Write-Output "Adding the Required Modules section"
+    $replaceRequiredModules = "  RequiredModules = @($dependencies)"
```
Comment (CodeCyclone) [inline scripts/UpdateMetaModule.ps1]: Looks like tabbing is off

File: scripts/UpdateMetaModule.ps1 (hunk 4 above)
Comment (CodeCyclone) [reply scripts/UpdateMetaModule.ps1]: You will also need to wrap dependency names into a string (wrap with ')

File: scripts/UpdateMetaModule.ps1 [hunk 6]
```diff
@@ -0,0 +1,82 @@
...
+    $psdFileContent = $psdFileContent -replace ' # RequiredModules = @[(][)]', $replaceRequiredModules
+    $updatePsdFile = $true
```
Comment (CodeCyclone) [inline scripts/UpdateMetaModule.ps1]: You don't need to use this variable, that was if you do a -match then replace pattern

File: scripts/UpdateMetaModule.ps1 (hunk 3 above)
Comment (speshoria) [reply scripts/UpdateMetaModule.ps1]: Removing both attributes together doesn't work. I added the following to remove xmlns path separately:
$nuspec.package.RemoveAttribute("xmlns")

File: scripts/UpdateMetaModule.ps1 (hunk 5 above)
Comment (speshoria) [reply scripts/UpdateMetaModule.ps1]: Done

File: scripts/UpdateMetaModule.ps1 (hunk 2 above)
Comment (speshoria) [reply scripts/UpdateMetaModule.ps1]: Done

File: scripts/UpdateMetaModule.ps1 (hunk 4 above)
Comment (speshoria) [reply scripts/UpdateMetaModule.ps1]: Done

File: scripts/UpdateMetaModule.ps1 (hunk 1 above)
Comment (speshoria) [reply scripts/UpdateMetaModule.ps1]: Done

**Pull request merged**